lib/rosette/serializers/xml-serializer.rb

```diff
@@ -0,0 +1,147 @@
+        super
+        @writer = XmlWriteStream.from_stream(stream)
+        writer.write_header(encoding: encoding.to_s)
+        after_initialize
+      end
+
+      def after_initialize
+      end
+
+      def self.default_extension
+        '.xml'
+      end
+
+      class AndroidSerializer < XmlSerializer
+        PLURAL_FORMS = %w(
+          zero one two few many other
+        )
+
+        attr_reader :plurals, :arrays
+
+        def after_initialize
+          writer.open_tag('resources')
+          @plurals = Hash.new { |hash, key| hash[key] = {} }
+          @arrays = Hash.new { |hash, key| hash[key] = {} }
+        end
+
+        def write_raw(text)
+          stream.write(text)
+        end
+
+        def write_key_value(key, value)
+          key_parts = split_key(key)
+
+          case detect_string_type(key_parts)
+            when :plural
+              record_plural(key_parts, value)
+            when :array_element
+              record_array_element(key_parts, value)
+            else
+              write_string(key, value)
+          end
+        end
+
+        def flush
+          write_recorded_plurals
+          write_recorded_arrays
+          writer.flush
+          stream.flush
+        end
+
+        protected
+
+        def split_key(key)
+          if idx = key.rindex('.')
+            [key[0..(idx - 1)], key[(idx + 1)..-1]]
+          else
+            # this case should never happen
+            [key, '']
+          end
+        end
+
+        def record_plural(key_parts, value)
+          plurals[key_parts.first][key_parts.last] = value
+        end
+
+        def record_array_element(key_parts, value)
+          arrays[key_parts.first][key_parts.last.to_i] = value
+        end
+
+        def write_plural(name, plural_forms)
+          writer.open_tag(:plurals, name: name)
+
+          plural_forms.each_pair do |quantity, value|
+            writer.open_tag(:item, quantity: quantity)
+            write_text(value)
+            writer.close_tag
+          end
+
+          writer.close_tag
+        end
+
+        def write_array(name, array_elements)
+          writer.open_tag(:'string-array', name: name)
+          count = array_elements.keys.max
+
+          (0..count).each do |i|
+            writer.open_tag(:item)
+            write_text(array_elements[i] || '')
+            writer.close_tag
+          end
+
+          writer.close_tag
+        end
+
+        def write_recorded_plurals
+          plurals.each_pair do |name, plural_forms|
+            write_plural(name, plural_forms)
+          end
+        end
+
+        def write_recorded_arrays
+          arrays.each_pair do |name, array_elements|
+            write_array(name, array_elements)
+          end
+        end
+
+        def write_string(key, value)
+          writer.open_tag(:string, name: key)
+          write_text(value)
+          writer.close_tag
+        end
+
+        def write_text(text)
+          escaped_text = text.gsub('"', '\"')
```

Comment:
Good question! Android XML works a little differently than normal XML. Escaping &, <, >, etc isn't necessary because string text can be enclosed in double quotes, eg:

```xml
<resources>
    <string name="MyString">"Alaska & Hawai'i"</string>
</resources>
```

All the text between the double quotes is essentially exempt from normal XML escaping rules. Because of this, all that needs to be escaped are any double quotes in the text. Here's a [complete description](http://developer.android.com/guide/topics/resources/string-resource.html) from Google.